libraries/Ethernet/src/EthernetClient.cpp

```diff
@@ -173,5 +180,5 @@ bool EthernetClient::operator==(const EthernetClient& rhs) {
 }
 
 uint8_t EthernetClient::getSocketNumber() {
-  return _sock;
+  return 0;
 }
```

Comment:
It is not a limitation because getSocketNumber() is not an Arduino function but implemented for the W5100 architecture. I just keep it for compatibility.